### PR TITLE
Add vault connector resource

### PR DESCRIPTION
**Pull request by rathodmeetsatish**

### CODE REVIEW

File: internal/service/platform/connector/vault.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceConnectorVault() *schema.Resource {
+	resource := &schema.Resource{
+		Description:   "Resource for creating a Hashicorp Vault Secret Manager connector.",
+		ReadContext:   resourceConnectorVaultRead,
+		CreateContext: resourceConnectorVaultCreateOrUpdate,
+		UpdateContext: resourceConnectorVaultCreateOrUpdate,
+		DeleteContext: resourceConnectorDelete,
```
Comment (rijajoo) [inline internal/service/platform/connector/vault.go:20]: rename to resourceConnectorVaultDelete.

**Comment by rathodmeetsatish** [comment]
trigger build

**Review by rijajoo — APPROVED** [review]

**Review by RajBaviskar — APPROVED** [review]

File: docs/data-sources/platform_connector_vault.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- `name` (String) Name of the resource.
+- `org_id` (String) Unique identifier of the organization.
+- `project_id` (String) Unique identifier of the project.
+
+### Read-Only
+
+- `access_type` (String) Access type.
+- `app_role_id` (String) ID of App Role.
+- `auth_token` (String) This is the authentication token for Vault.
+- `aws_region` (String) This is the Aws region where aws iam auth will happen.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:36]: (String) AWS region where the AWS IAM authentication will happen.
AWS IAM

File: docs/data-sources/platform_connector_vault.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- `base_path` (String) This is the location of the Vault directory where Secret will be stored.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:37]: (String) Location of the Vault directory where the secret will be stored.

File: docs/data-sources/platform_connector_vault.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+- `identifier` (String) Unique identifier of the resource.
+- `name` (String) Name of the resource.
+- `org_id` (String) Unique identifier of the organization.
+- `project_id` (String) Unique identifier of the project.
+
+### Read-Only
+
+- `access_type` (String) Access type.
+- `app_role_id` (String) ID of App Role.
+- `auth_token` (String) This is the authentication token for Vault.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:35]: (String) Authentication token for Vault.

File: docs/data-sources/platform_connector_vault.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+- `aws_region` (String) This is the Aws region where aws iam auth will happen.
+- `base_path` (String) This is the location of the Vault directory where Secret will be stored.
+- `default` (Boolean) Is default or not.
+- `delegate_selectors` (Set of String) List of Delegate Selectors that belong to the same Delegate and are used to connect to the Secret Manager.
+- `description` (String) Description of the resource.
+- `id` (String) The ID of this resource.
+- `is_default` (Boolean) Is default or not.
+- `is_read_only` (Boolean) Read only or not.
+- `k8s_auth_endpoint` (String) This is the path where kubernetes auth is enabled in Vault.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:44]: Kubernetes

File: docs/data-sources/platform_connector_vault.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- `namespace` (String) This is the Vault namespace where Secret will be created.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:45]: Vault namespace where the Secret will be created.

File: docs/data-sources/platform_connector_vault.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+- `read_only` (Boolean) Read only.
+- `renew_app_role_token` (Boolean) Boolean value to indicate if appRole token renewal is enabled or not.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:47]: AppRole

File: docs/data-sources/platform_connector_vault.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- `renewal_interval_minutes` (Number) This is the time interval for token renewal.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:48]: The time interval for the token renewal.

File: docs/data-sources/platform_connector_vault.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+- `secret_engine_manually_configured` (Boolean) Manually entered Secret Engine.
+- `secret_engine_name` (String) Name of the Secret Engine.
+- `secret_engine_version` (Number) Version of Secret Engine.
+- `secret_id` (String) ID of the Secret.
+- `service_account_token_path` (String) This is the SA token path where the token is mounted in the K8s Pod.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:53]: The Service Account token path in the K8s pod where the token is mounted.

File: docs/data-sources/platform_connector_vault.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+- `sink_path` (String) This is the location at which auth token is to be read from.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:54]: (String) The location from which the authentication token should be read.

File: docs/data-sources/platform_connector_vault.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+- `tags` (Set of String) Tags to associate with the resource. Tags should be in the form `name:value`.
+- `use_aws_iam` (Boolean) Boolean value to indicate if Aws Iam is used for authentication.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:56]: AWS IAM

File: docs/data-sources/platform_connector_vault.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+- `use_k8s_auth` (Boolean) Boolean value to indicate if K8s Auth is used for authentication.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:57]: Why is the casing different? In some places, it is "auth" and in some, it is "Auth". Please make it consistent throughout.

File: docs/data-sources/platform_connector_vault.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+- `use_vault_agent` (Boolean) Boolean value to indicate if Vault Agent is used for authentication.
+- `vault_aws_iam_role` (String) This is the Vault role defined to bind to aws iam account/role being accessed.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_vault.md:59]: Please correct the casing to AWS IAM everywhere.

File: docs/resources/platform_connector_vault.md [hunk 14]
```diff
@@ -0,0 +1,192 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_connector_vault Resource - terraform-provider-harness"
+subcategory: "Next Gen"
+description: |-
+  Resource for creating a Hashicorp Vault Secret Manager connector.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_vault.md:6]: HashiCorp
This needs to be corrected everywhere.

File: docs/resources/platform_connector_vault.md [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `identifier` (String) Unique identifier of the resource.
+- `name` (String) Name of the resource.
+- `renewal_interval_minutes` (Number) This is the time interval for token renewal.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_vault.md:146]: We can start the sentence like this - "The time interval for the token renewal" and remove"This is"
This needs to be followed throughout. Refer to earlier comments.

File: docs/resources/platform_connector_vault.md [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+- `vault_url` (String) URL of the HashiCorp Vault.
+
+### Optional
+
+- `access_type` (String) Access type.
+- `app_role_id` (String) ID of App Role.
+- `auth_token` (String) This is the authentication token for Vault.
+- `aws_region` (String) This is the Aws region where aws iam auth will happen.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_vault.md:154]: AWS
IAM

File: internal/service/platform/connector/vault_data_source.go [hunk 17]
```diff
@@ -0,0 +1,156 @@
+package connector
+
+import (
+	"github.com/harness/terraform-provider-harness/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
```
Comment (rashminandasahoo) [inline internal/service/platform/connector/vault_data_source.go:7]: Refer to previous comments for this section too.

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

**Pull request merged**